docs/gelly_guide.md

```diff
@@ -0,0 +1,425 @@
+
+For example, assume that you want to select the minimum weight of all out-edges for each vertex in the following graph:
+
+<p class="text-center">
+    <img alt="reduceOnEdges Example" width="50%" src="img/gelly-example-graph.png"/>
+</p>
+
+The following code will collect the out-edges for each vertex and apply the `SelectMinWeight()` user-defined function on each of the resulting neighborhoods:
+
+{% highlight java %}
+Graph<Long, Long, Double> graph = ...
+
+DataSet<Tuple2<Long, Double>> minWeights = graph.reduceOnEdges(
+				new SelectMinWeight(), EdgeDirection.OUT);
+
+// user-defined function to select the minimum weight
+static final class SelectMinWeight implements EdgesFunction<Long, Double, Tuple2<Long, Double>> {
+
+    public Tuple2<Long, Double> iterateEdges(Iterable<Tuple2<Long, Edge<Long, Double>>> edges) {
+
+        long minWeight = Double.MAX_VALUE;
+        long vertexId = -1;
+
+        for (Tuple2<Long, Edge<Long, Double>> edge: edges) {
+            if (edge.f1.getValue() < weight) {
+            weight = edge.f1.getValue();
+            vertexId = edge.f0;
+        }
+        return new Tuple2<Long, Double>(vertexId, minWeight);
+    }
+}
+{% endhighlight %}
+
+<p class="text-center">
+    <img alt="reduceOnEdges Example" width="50%" src="img/gelly-reduceOnEdges.png"/>
+</p>
+
+Similarly, assume that you would like to compute the sum of the values of all in-coming neighbors, for every vertex. The following code will collect the in-coming neighbors for each vertex and apply the `SumValues()` user-defined function on each neighborhood:
+
+{% highlight java %}
+Graph<Long, Long, Double> graph = ...
+
+DataSet<Tuple2<Long, Long>> verticesWithSum = graph.reduceOnNeighbors(
+				new SumValues(), EdgeDirection.IN);
+
+// user-defined function to sum the neighbor values
+static final class SumValues implements NeighborsFunction<Long, Long, Long, Tuple2<Long, Long>> {
+		
+	public Tuple2<Long, Long> iterateNeighbors(Iterable<Tuple3<Long, Edge<Long, Long>, 
+		Vertex<Long, Long>>> neighbors) {
+		
+		long sum = 0;
+		long vertexId = -1;
+
+		for (Tuple3<Long, Edge<Long, Long>, Vertex<Long, Long>> neighbor : neighbors) {
+			vertexId = neighbor.f0;
+			sum += neighbor.f2.getValue();
+		}
+		return new Tuple2<Long, Long>(vertexId, sum);
+	}
+}
+{% endhighlight %}
+
+<p class="text-center">
+    <img alt="reduseOnNeighbors Example" width="70%" src="img/gelly-reduceOnNeighbors.png"/>
+</p>
+
+When the aggregation computation does not require access to the vertex value, it is advised to use the more efficient `EdgesFunction` and `NeighborsFunction` for the user-defined functions. When access to the vertex value is required, one should use `EdgesFunctionWithVertexValue` and `NeighborsFunctionWithVertexValue` instead. 
```

Comment:
I was a little bit confused which vertex value you meant, because in the example above we are accessing the vertex values of the neighbours. Maybe we could emphasize that we're talking here about the vertex for which the aggregation is performed.